polynote-frontend/polynote/cell.js

```diff
@@ -140,22 +140,24 @@ export class Cell extends UIEventTarget {
         // TODO: some way to display the KeyMap to users
         this.keyMap = new Map([
             [KeyPress.of("ArrowUp").h, (pos, range, selection, cell) => {
-                if (!selection && pos.lineNumber === range.startLineNumber && pos.column === range.startColumn) {
+                if (!selection && pos.lineNumber <= range.startLineNumber && pos.column <= range.startColumn) {
                     cell.dispatchEvent(new AdvanceCellEvent(cell.id, true));
 
                 }
             }],
             [KeyPress.of("ArrowDown").h, (pos, range, selection, cell) => {
-                if (!selection && pos.lineNumber === range.endLineNumber && pos.column === range.endColumn) {
+                if (!selection && pos.lineNumber >= range.endLineNumber && pos.column >= range.endColumn) {
```

Comment:
In vim normal mode the cursor is a block, so it never selects the end of a line (i.e., it selects the last character in the line, not the location AFTER that character like a regular cursor). This is compensated for here: https://github.com/polynote/polynote/blob/master/polynote-frontend/polynote/cell.js#L345-L347

However this compensation backfired when the cell was empty because there was only one possible position. The vim position was modified to be after the "end" of the range. So the fix is to basically say that if the position is outside the bounds (in the right direction), we treat it as at the limit of the bound.